client.go

```diff
@@ -924,6 +924,55 @@ func (c *Client) MkdirAll(path string) error {
 	return nil
 }
 
+// DeleteAllResources delete files recursively in the directory and Recursively delete subdirectories.
+// An error will be returned if no file or directory with the specified path exists
+func (c *Client) DeleteAllResources(filePath string) error {
+
+	// Get the file/directory information
+	fileInfo, err := c.Stat(filePath)
+	if err != nil {
+		return err
+	}
+
+	if fileInfo.IsDir() {
+		// Delete files recursively in the directory
+		files, err := c.ReadDir(filePath)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				// Recursively delete subdirectories
+				err = c.DeleteAllResources(filePath + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			} else {
+				// Delete individual files
+				err = c.Remove(filePath + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		// Delete the empty directory
+		err = c.RemoveDirectory(filePath)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Delete individual files
+		err = c.Remove(filePath)
+		if err != nil {
+			return err
+		}
```

Comment:
Same as above, this would be more clear as simply `return c.Remove(path)`.

Since the `if` block will end in a `return`, this block does not need to be wrapped in an `else` block.

Once this isn’t wrapped in an `else` block, the `return c.RemoveDirectory(path)` is behaviorally the same as this new ending: `return c.Remove(path)`, that is, we could just fall out of the `if` block and let the `Remove` take care of removing both the file and the directory.

So, the code should look more like:
```
if fi.IsFile() { // `fi` is a well-known abbreviation for a `FileInfo` variable.
	files, err := c.ReadDir(path)
	…
	for _, file := range files {
		if file.IsDir() {
			…
		} else {
			…
		}
	}
}
return c.Remove(path)
```